cmd/validate.go

```diff
@@ -66,6 +79,60 @@ this command.
 	},
 }
 
+func validate(ctx context.Context, entity interface{}, kongClient *kong.Client, entityType string) string {
+	_, err := performValidate(ctx, kongClient, entity, entityType)
+	output := ""
+	if err != nil {
+		output = color.New(color.FgRed, color.Bold).Sprintf("%s: %v", entityType, err)
+	}
+	return output
+}
+
+func validateEntities(ctx context.Context, obj interface{}, kongClient *kong.Client, entityType string) {
+	// call GetAll on entity
+	method := reflect.ValueOf(obj).MethodByName("GetAll")
+	entities := method.Call([]reflect.Value{})[0].Interface()
+
+	values := reflect.ValueOf(entities)
+	var wg sync.WaitGroup
+	wg.Add(values.Len())
+	for i := 0; i < values.Len(); i++ {
+		go func(i int) {
+			defer wg.Done()
+			output := validate(ctx, values.Index(i).Interface(), kongClient, entityType)
+			if output != "" {
+				fmt.Println(output)
+			}
+		}(i)
+	}
+	wg.Wait()
+}
+
+func validateWithKong(cmd *cobra.Command, ks *state.KongState) error {
+	ctx := cmd.Context()
+	kongClient, err := utils.GetKongClient(rootConfig)
+	if err != nil {
+		return err
+	}
+	validateEntities(ctx, ks.Services, kongClient, "services")
+	validateEntities(ctx, ks.ACLGroups, kongClient, "acls")
+	validateEntities(ctx, ks.BasicAuths, kongClient, "basicauth_credentials")
+	validateEntities(ctx, ks.CACertificates, kongClient, "ca_certificates")
+	validateEntities(ctx, ks.Certificates, kongClient, "certificates")
+	validateEntities(ctx, ks.Consumers, kongClient, "consumers")
+	validateEntities(ctx, ks.Documents, kongClient, "documents")
+	validateEntities(ctx, ks.HMACAuths, kongClient, "hmacauth_credentials")
+	validateEntities(ctx, ks.JWTAuths, kongClient, "jwt_secrets")
+	validateEntities(ctx, ks.KeyAuths, kongClient, "keyauth_credentials")
+	validateEntities(ctx, ks.Oauth2Creds, kongClient, "oauth2_credentials")
+	validateEntities(ctx, ks.Plugins, kongClient, "plugins")
+	validateEntities(ctx, ks.Routes, kongClient, "routes")
+	validateEntities(ctx, ks.SNIs, kongClient, "snis")
+	validateEntities(ctx, ks.Targets, kongClient, "targets")
+	validateEntities(ctx, ks.Upstreams, kongClient, "upstreams")
```

Comment:
RBAC resources are missing here. Can you include those?

Reply:
Ops, missed this...what is/are the db entities associated to RBAC resources? I couldn't find them :\